api/types/access_request.go

```diff
@@ -776,6 +781,28 @@ func (f *AccessRequestFilter) FromMap(m map[string]string) error {
 
 // Match checks if a given access request matches this filter.
 func (f *AccessRequestFilter) Match(req AccessRequest) bool {
+	// only return if the request was made by the api requester
+	if f.Scope == AccessRequestScope_MY_REQUESTS && req.GetUser() != f.Requester {
+		return false
+	}
+	// a user cannot review their own requests
+	if f.Scope == AccessRequestScope_NEEDS_REVIEW && req.GetUser() == f.Requester {
+		return false
+	}
+	// only match if the api requester has submit a review
+	if f.Scope == AccessRequestScope_REVIEWED {
+		reviews := req.GetReviews()
+		var reviewers []string
```

Comment:
since we already know the length
```suggestion
		reviewers := make([]string, len(reviews))
```

Reply:
I don't think we know the length. Requests can have multiple reviewers